src/test.js

```diff
@@ -30,13 +30,24 @@ export default function Test( settings ) {
 	++Test.count;
 
 	this.expected = null;
-	extend( this, settings );
 	this.assertions = [];
 	this.semaphore = 0;
 	this.module = config.currentModule;
 	this.stack = sourceFromStacktrace( 3 );
 	this.steps = [];
 
+	// If a module is skipped, all its tests and the tests of the child suites
+	// should be treated as skipped even if they are defined as `only` or `todo`.
```

Comment:
I'm unsure if we want to do this for `only` tests. I can see a scenario where a module is skipped, but then they want to test an individual test without re-enabling the whole suite.

I tend to think there needs to be some order of precedence for the modifiers. I think `only > skip > todo > test`. Does that make sense?